### ISSUE TITLE
Error whenever n_jobs > 1

### ISSUE THREAD

**Issue by rdenham** [jgrss/zones#1]
If I set n_jobs to be more than 1 I get an error:

```
Traceback (most recent call last):
  File "/home/robert/miniconda3/envs/uav/lib/python3.6/site-packages/joblib/externals/loky/process_executor.py", line 418, in _process_worker
    r = call_item()
  File "/home/robert/miniconda3/envs/uav/lib/python3.6/site-packages/joblib/externals/loky/process_executor.py", line 272, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/home/robert/miniconda3/envs/uav/lib/python3.6/site-packages/joblib/_parallel_backends.py", line 608, in __call__
    return self.func(*args, **kwargs)
  File "/home/robert/miniconda3/envs/uav/lib/python3.6/site-packages/joblib/parallel.py", line 256, in __call__
    for func, args, kwargs in self.items]
  File "/home/robert/miniconda3/envs/uav/lib/python3.6/site-packages/joblib/parallel.py", line 256, in <listcomp>
    for func, args, kwargs in self.items]
  File "/home/robert/.local/lib/python3.6/site-packages/zones/raster_stats.py", line 359, in worker
    image_array[null_idx] = no_data
IndexError: index 58 is out of bounds for axis 0 with size 1
```
Here is as an example (in ipython)
```
import zones
rast, vect = zones.datasets.load_01_single()
rast2 = "/tmp/raster_test.tif"
!gdalwarp -tr 1 1 {rast} {rast2}
zs = zones.RasterStats(rast2,
                       vect,
                       verbose=2,
                       n_jobs=2,
                       band=1)
gdf = zs.calculate('mean')
```
Note that this doesn't happen if you use the original test data, but I think that is because it is so small that no parallel stuff is happening. So I increased the size of the image by resampling.

**Comment by jgrss** [comment]
There was no check for multi-dimensional arrays when indexing null values. The check existed in the single core process, but not in the `joblib` process.

**Comment by jgrss** [comment]
Fixed in [1ebd72c4ca6ddf854a00cb716263636c1c0a20e5](https://github.com/jgrss/zones/commit/1ebd72c4ca6ddf854a00cb716263636c1c0a20e5)

**Comment by rdenham** [comment]
Thanks, tested it on the small example, works well. I'll test it now on the larger example.

**Comment by rdenham** [comment]
Might be a problem with the multiband, the following code fails too:

```
import zones
rast, vect = zones.datasets.load_01_multi()
rast2 = "/tmp/raster_test.tif"
!gdalwarp -tr 1 1 {rast} {rast2}
zs = zones.RasterStats(rast2,
                       vect,
                       verbose=2,
                       n_jobs=2,
                       band=1)
gdf = zs.calculate('mean')
```

**Comment by jgrss** [comment]
The problem was that single-band arrays were left as 3d instead of being squeezed to 2d. Fixed in [5e11b0c0778d51d188931bc0532a2dd469572250](https://github.com/jgrss/zones/commit/5e11b0c0778d51d188931bc0532a2dd469572250)
